cli/yamls.py

```diff
@@ -141,12 +141,29 @@ def replace_lookup(self):
                 break
 
             for a_lookup in lookups:
-                lookup_key = re.search('(\w+\.?)+ *?\}\}', a_lookup)
-                lookup_key = lookup_key.group(0).strip()[:-2].strip()
-                lookup_value = self.dict_lookup(lookup_key.split("."))
-
-                if isinstance(lookup_value, Lookup):
-                    return
+                visited_lookups = []
+                lookup_value = a_lookup
+                while True:
+                    lookup_key = re.search('(\w+\.?)+ *?\}\}', a_lookup)
+                    if lookup_key is None:
+                        # if key cannot be found in lookup consider
+                        # it as a value and exit
+                        break
+
+                    lookup_key = lookup_key.group(0).strip()[:-2].strip()
+                    # check if lookups are in circular reference
+                    if lookup_key in visited_lookups:
+                        raise exceptions.IRInfiniteLookupException(
+                            visited_lookups)
+                    visited_lookups.append(lookup_key)
+                    lookup_value = self.dict_lookup(lookup_key.split("."))
+
+                    if not isinstance(lookup_value, Lookup):
```

Comment:
The entire 'if' block is not needed.
We're no longer support old style lookups ('!lookup key.subkey') only new style lookups ('{{!lookup key.subkey}}') so there will be no Look objects...